src/libpriv/rpmostree-scripts.c

```diff
@@ -259,19 +276,43 @@ run_script_in_bwrap_container (int rootfs_fd,
   if (!bwrap)
     goto out;
 
-  if (stdin_fd >= 0)
-    rpmostree_bwrap_set_child_setup (bwrap, script_child_setup_stdin, GINT_TO_POINTER (stdin_fd));
+
+  struct ChildSetupData data = { .stdin_fd = stdin_fd };
+
+  const char *id = glnx_strjoina ("rpm-ostree(", pkg_script, ")");
+  data.stdout_fd = stdout_fd = sd_journal_stream_fd (id, LOG_INFO, 0);
+  if (stdout_fd < 0)
+    {
+      glnx_throw_errno_prefix (error, "While creating stdout stream fd");
+      goto out;
+    }
+
+  data.stderr_fd = stderr_fd = sd_journal_stream_fd (id, LOG_ERR, 0);
+  if (stderr_fd < 0)
+    {
+      glnx_throw_errno_prefix (error, "While creating stderr stream fd");
+      goto out;
+    }
+
+  data.all_fds_initialized = TRUE;
+  rpmostree_bwrap_set_child_setup (bwrap, script_child_setup, &data);
 
   rpmostree_bwrap_append_child_argv (bwrap,
                                      interp,
                                      postscript_path_container,
                                      script_arg,
                                      NULL);
 
-  if (!rpmostree_bwrap_run (bwrap, error))
-    goto out;
+  ret = rpmostree_bwrap_run (bwrap, error);
+
+  if (!ret && error)
```

Comment:
How about instead:

```
if (!rpmostree_bwrap_run (...))
  {
     ...error prefixing...
     goto out;
  }

ret = TRUE;
out:
  ...
```

? Feels like a closer match for what I think of as a C "try/catch" flow.